### PR TITLE
add speedyspeech model and example with baker dataset.

### PR DESCRIPTION
**Pull request by iclementine**
[=] SpeedySpeech Model;
[=] Preprocessing pipelines for baker dataset;
[=] SpeedySpeech Updater, main training loop;
[=] Evaluator;
[=] whole training script and inference;
[=] End to end inference with PWG & Text analysis;
[ ] Inference ModelExport.

### CODE REVIEW

File: examples/parallelwave_gan/baker/run.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FLAGS_cudnn_exhaustive_search=true \
+FLAGS_conv_workspace_size_limit=4000 \
```
Comment (zh794390558) [inline examples/parallelwave_gan/baker/run.sh:2]: 这个有什么用？

File: examples/parallelwave_gan/baker/run.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+CUDA_VISIBLE_DEVICES="3" \
```
Comment (zh794390558) [inline examples/parallelwave_gan/baker/run.sh]: gpuid 可以从外部指定

File: examples/speedyspeech/baker/compute_statistics.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    # calculate statistics
+    scaler = StandardScaler()
+    for datum in tqdm(dataset):
+        # StandardScalar supports (*, num_features) by default
+        scaler.partial_fit(datum[args.field_name])
+
+    stats = np.stack([scaler.mean_, scaler.scale_], axis=0)
+    np.save(
+        os.path.join(args.dumpdir, "stats.npy"),
```
Comment (zh794390558) [inline examples/speedyspeech/baker/compute_statistics.py]: 最好是加到args上

File: examples/parallelwave_gan/baker/run.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+python train.py \
+    --train-metadata=dump/train/norm/metadata.jsonl \
+    --dev-metadata=dump/dev/norm/metadata.jsonl \
+    --config=conf/debug.yaml \
```
Comment (zh794390558) [inline examples/parallelwave_gan/baker/run.sh]: 没有debug.yaml这个文件

File: examples/speedyspeech/baker/conf/default.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+###########################################################
+#                FEATURE EXTRACTION SETTING               #
+###########################################################
+sr: 24000     # Sampling rate.
+n_fft: 2048           # FFT size.
+hop_length: 300            # Hop size.
```
Comment (zh794390558) [inline examples/speedyspeech/baker/conf/default.yaml:6]: unit in sample

File: examples/speedyspeech/baker/conf/default.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+
+
+###########################################################
+#                       MODEL SETTING                     #
+###########################################################
+model:
+  vocab_size: 101 # 99 + 2
+  tone_size: 8 # 6 + 2
```
Comment (zh794390558) [inline examples/speedyspeech/baker/conf/default.yaml:34]: +2是什么？为什么不是5呢？

File: examples/parallelwave_gan/baker/run.sh (hunk 1 above)
Comment (iclementine) [reply examples/parallelwave_gan/baker/run.sh:2]: 这个就是为了加快 pwg 的训练

File: examples/parallelwave_gan/baker/run.sh (hunk 2 above)
Comment (iclementine) [reply examples/parallelwave_gan/baker/run.sh]: ok，那删除这一行

File: examples/speedyspeech/baker/compute_statistics.py (hunk 3 above)
Comment (iclementine) [reply examples/speedyspeech/baker/compute_statistics.py]: 把什么加到 args 上？

File: examples/parallelwave_gan/baker/run.sh (hunk 4 above)
Comment (iclementine) [reply examples/parallelwave_gan/baker/run.sh]: 我修改一下 default.yaml

File: examples/speedyspeech/baker/conf/default.yaml (hunk 6 above)
Comment (iclementine) [reply examples/speedyspeech/baker/conf/default.yaml:34]: <unk> and <pad> 这里没有使用 <s> 和 </s>
声调数是 6 是因为 1， 2， 3， 4， 5 以及一个没有声调的 0.

File: examples/speedyspeech/baker/normalize.py [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+    # restore scaler
+    scaler = StandardScaler()
+    scaler.mean_ = np.load(args.stats)[0]
+    scaler.scale_ = np.load(args.stats)[1]
+
+    # from version 0.23.0, this information is needed
+    scaler.n_features_in_ = scaler.mean_.shape[0]
+
+    with open("phones.txt", 'rt') as f:
```
Comment (zh794390558) [inline examples/speedyspeech/baker/normalize.py]: args 中指定会合理

File: examples/speedyspeech/baker/normalize.py [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+        phones = [line.strip() for line in f.readlines()]
+
+    with open("tones.txt", 'rt') as f:
```
Comment (zh794390558) [inline examples/speedyspeech/baker/normalize.py]: args 中指定会合理

File: examples/speedyspeech/baker/preprocess.py [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+        window=window,
+        pad_mode="reflect")
+    spc = np.abs(x_stft)  # (#bins, #frames,)
+
+    # get mel basis
+    fmin = 0 if fmin is None else fmin
+    fmax = sr / 2 if fmax is None else fmax
+    mel_basis = librosa.filters.mel(sr, n_fft, n_mels, fmin, fmax)
+
+    return np.log10(np.maximum(eps, np.dot(mel_basis, spc)))
```
Comment (zh794390558) [inline examples/speedyspeech/baker/preprocess.py:92]: 一般用 ln

File: examples/speedyspeech/baker/speedyspeech_updater.py [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+from paddle.nn import functional as F
+from paddle.fluid.layers import huber_loss
+
+from parakeet.modules.ssim import ssim
```
Comment (zh794390558) [inline examples/speedyspeech/baker/speedyspeech_updater.py:19]: 没有找到ssim的模块

File: examples/speedyspeech/baker/speedyspeech_updater.py [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+                predicted_durations, paddle.log(target_durations), delta=1.0),
+            text_mask, )
+
+        # ssim loss
+        ssim_loss = 1.0 - ssim((decoded * spec_mask).unsqueeze(1),
+                               (target_mel * spec_mask).unsqueeze(1))
+
+        loss = l1_loss + ssim_loss + duration_loss
+
+        # import pdb; pdb.set_trace()
```
Comment (zh794390558) [inline examples/speedyspeech/baker/speedyspeech_updater.py:105]: remove this line

File: examples/speedyspeech/baker/synthesize.sh [hunk 12]
```diff
@@ -0,0 +1,10 @@
+python synthesize.py \
+  --speedyspeech-config=conf/default.yaml \
+  --speedyspeech-checkpoint=exp/debug/checkpoints/snapshot_iter_91800.pdz \
```
Comment (zh794390558) [inline examples/speedyspeech/baker/synthesize.sh:3]: ckpt需要外部指定

File: examples/speedyspeech/baker/compute_statistics.py (hunk 3 above)
Comment (zh794390558) [reply examples/speedyspeech/baker/compute_statistics.py]: stats.npy 这个最好加上

**Review by zh794390558 — COMMENTED** [review]
LGTM

File: examples/speedyspeech/baker/speedyspeech_updater.py (hunk 11 above)
Comment (iclementine) [reply examples/speedyspeech/baker/speedyspeech_updater.py:105]: done

File: examples/speedyspeech/baker/preprocess.py (hunk 9 above)
Comment (iclementine) [reply examples/speedyspeech/baker/preprocess.py:92]: 这个是为了和 PWG 的一致性。后续可以重新用 log(e) 的重新训练

File: examples/speedyspeech/baker/normalize.py (hunk 7 above)
Comment (iclementine) [reply examples/speedyspeech/baker/normalize.py]: 已经添加

File: examples/speedyspeech/baker/compute_statistics.py (hunk 3 above)
Comment (iclementine) [reply examples/speedyspeech/baker/compute_statistics.py]: done

**Review by zh794390558 — APPROVED** [review]
LGTM
